Backend/app/rider-platform/rider-app/Main/src/API/UI/Select.hs

```diff
@@ -73,15 +81,34 @@ type API =
 
 handler :: FlowServer API
 handler =
-  select2
+  select
+    :<|> select2
     :<|> selectList
     :<|> selectResult
     :<|> cancelSearch
 
+select :: (Id DPerson.Person, Id Merchant.Merchant) -> Id DEstimate.Estimate -> DSelect.DSelectReq -> FlowHandler DSelect.DSelectResultRes
+select (personId, _) estimateId req = withFlowHandlerAPI . withPersonIdLogTag personId $ do
+  dSelectReq <- DSelect.select personId estimateId req
```

Comment:
This statement should be inside Redis Lock

Reply:
but i want to use a parameter `autoAssignEnabled` which is in `dSelectReq`. so if i keep it in redis lock i can't use that outside.